### PR TITLE
[GCU]Remove GCU unique lane check for duplicate lanes platforms

### PR DESCRIPTION
**Pull request by wen587**
#### What I did
Remove unique lane check for specific platforms to unblock GCU nightly test.
#### How I did it
Add platform check for unique lane validator
#### How to verify it
Run unit test

### CODE REVIEW

**Comment by wen587** [comment]
"DPB_1_TO_4__SUCCESS" in patch_sorter_test  will fail after unique lane check removed.
Temporary remove it to pass unit test.


Details below:
After removing the unique lane check, it directly adds duplicate lanes in Ethernet1~3 before removing lanes in Ethernet0.
I doubt if it is a valid move.

Expected change involving unique lane check
```
[
    [{"op": "replace", "path": "/PORT/Ethernet0/alias", "value": "Eth1/1"}], 
    [{"op": "replace", "path": "/PORT/Ethernet0/description", "value": ""}], 
    [{"op": "replace", "path": "/PORT/Ethernet0/speed", "value": "10000"}], 
    [{"op": "remove", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports"}], 
    [{"op": "remove", "path": "/ACL_TABLE"}], 
    [{"op": "add", "path": "/ACL_TABLE", "value": {"NO-NSW-PACL-V4": {"type": "L3"}}}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports", "value": ["Ethernet0"]}], 
    [{"op": "remove", "path": "/VLAN_MEMBER"}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/policy_desc", "value": "NO-NSW-PACL-V4"}], 
    [{"op": "remove", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports"}], [{"op": "remove", "path": "/PORT"}], 
    [{"op": "add", "path": "/PORT", "value": {"Ethernet0": {"alias": "Eth1/1", "lanes": "65", "description": "", "speed": "10000"}}}], 
    [{"op": "add", "path": "/PORT/Ethernet3", "value": {"alias": "Eth1/4", "lanes": "68", "description": "", "speed": "10000"}}], 
    [{"op": "add", "path": "/PORT/Ethernet2", "value": {"alias": "Eth1/3", "lanes": "67", "description": "", "speed": "10000"}}], 
    [{"op": "add", "path": "/PORT/Ethernet1", "value": {"alias": "Eth1/2", "lanes": "66", "description": "", "speed": "10000"}}], 
    [{"op": "add", "path": "/VLAN_MEMBER", "value": {"Vlan100|Ethernet0": {"tagging_mode": "untagged"}}}], 
    [{"op": "add", "path": "/VLAN_MEMBER/Vlan100|Ethernet1", "value": {"tagging_mode": "untagged"}}], 
    [{"op": "add", "path": "/VLAN_MEMBER/Vlan100|Ethernet3", "value": {"tagging_mode": "untagged"}}], 
    [{"op": "add", "path": "/VLAN_MEMBER/Vlan100|Ethernet2", "value": {"tagging_mode": "untagged"}}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports", "value": ["Ethernet0"]}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports/1", "value": "Ethernet1"}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports/2", "value": "Ethernet2"}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports/3", "value": "Ethernet3"}]
]
```

Actual change after remove unique lane check:
```
[
    [{"op": "add", "path": "/PORT/Ethernet3", "value": {"alias": "Eth1/4", "lanes": "68", "description": "", "speed": "10000"}}], 
    [{"op": "add", "path": "/PORT/Ethernet2", "value": {"alias": "Eth1/3", "lanes": "67", "description": "", "speed": "10000"}}], 
    [{"op": "add", "path": "/PORT/Ethernet1", "value": {"alias": "Eth1/2", "lanes": "66", "description": "", "speed": "10000"}}], 
    [{"op": "add", "path": "/VLAN_MEMBER/Vlan100|Ethernet1", "value": {"tagging_mode": "untagged"}}], 
    [{"op": "add", "path": "/VLAN_MEMBER/Vlan100|Ethernet3", "value": {"tagging_mode": "untagged"}}], 
    [{"op": "add", "path": "/VLAN_MEMBER/Vlan100|Ethernet2", "value": {"tagging_mode": "untagged"}}], 
    [{"op": "replace", "path": "/PORT/Ethernet0/alias", "value": "Eth1/1"}], 
    [{"op": "replace", "path": "/PORT/Ethernet0/description", "value": ""}], 
    [{"op": "replace", "path": "/PORT/Ethernet0/speed", "value": "10000"}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports/1", "value": "Ethernet1"}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports/2", "value": "Ethernet2"}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports/3", "value": "Ethernet3"}], 
    [{"op": "remove", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports/0"}], 
    [{"op": "remove", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports"}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports", "value": ["Ethernet1"]}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports/1", "value": "Ethernet2"}], 
    [{"op": "remove", "path": "/VLAN_MEMBER/Vlan100|Ethernet0"}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports/0", "value": "Ethernet0"}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports/3", "value": "Ethernet3"}], 
    [{"op": "remove", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports/0"}], 
    [{"op": "remove", "path": "/PORT/Ethernet0"}], 
    [{"op": "add", "path": "/PORT/Ethernet0", "value": {"alias": "Eth1/1", "lanes": "65", "description": "", "speed": "10000"}}], 
    [{"op": "add", "path": "/VLAN_MEMBER/Vlan100|Ethernet0", "value": {"tagging_mode": "untagged"}}], 
    [{"op": "add", "path": "/ACL_TABLE/NO-NSW-PACL-V4/ports/0", "value": "Ethernet0"}]
]

```

File: tests/generic_config_updater/patch_sorter_test.py [hunk 1]
```diff
@@ -3034,6 +3034,9 @@ def test_patch_sorter_success(self):
         data = Files.PATCH_SORTER_TEST_SUCCESS
         skip_exact_change_list_match = False
         for test_case_name in data:
+            # DPB_1_TO_4__SUCCESS will fail when unique_lanes is set to False
```
Comment (qiluo-msft) [inline tests/generic_config_updater/patch_sorter_test.py:3037]: >False [](http://example.com/codeflow?start=72&length=5)

True ?

File: generic_config_updater/gu_common.py [hunk 2]
```diff
@@ -124,16 +126,17 @@ def validate_config_db_config(self, config_db_as_json):
 
             sy.validate_data_tree()
 
-            for supplemental_yang_validator in supplemental_yang_validators:
-                success, error = supplemental_yang_validator(config_db_as_json)
+            for supplemental_yang_validator, validator_params in \
+                    zip(supplemental_yang_validators, validators_params):
+                success, error = supplemental_yang_validator(*validator_params)
```
Comment (qiluo-msft) [inline generic_config_updater/gu_common.py:131]: >supplemental_yang_validator [](http://example.com/codeflow?start=33&length=27)

To make code more readable, you may use lambda in `supplemental_yang_validators`, and no need to `zip` with another array `validators_params`.

File: tests/generic_config_updater/patch_sorter_test.py (hunk 1 above)
Comment (qiluo-msft) [inline tests/generic_config_updater/patch_sorter_test.py:3037]: >fail [](http://example.com/codeflow?start=39&length=4)

Why one less validator will make original passing testcase fail?

Comment (wen587) [reply tests/generic_config_updater/patch_sorter_test.py:3037]: You can check above sorting output here https://github.com/sonic-net/sonic-utilities/pull/2343#issuecomment-1232568895 
When lane validator is removed, the sorting order is changed. That's why it fails.

File: generic_config_updater/gu_common.py [hunk 3]
```diff
@@ -116,6 +116,8 @@ def validate_config_db_config(self, config_db_as_json):
         # TODO: Move these validators to YANG models
         supplemental_yang_validators = [self.validate_bgp_peer_group,
                                         self.validate_lanes]
```
Comment (wen587) [inline generic_config_updater/gu_common.py:118]:
```suggestion
                                        lambda c: return self.validate_lanes(c, False)]
```

File: generic_config_updater/gu_common.py [hunk 4]
```diff
@@ -133,7 +138,7 @@ def validate_config_db_config(self, config_db_as_json):
 
         return True, None
 
-    def validate_lanes(self, config_db):
+    def validate_lanes(self, config_db, dup_lanes_platforms=[]):
```
Comment (qiluo-msft) [inline generic_config_updater/gu_common.py:141]: >[] [](http://example.com/codeflow?start=60&length=2)

This is a common bug. ref: [https://stackoverflow.com/a/44272023/2514803](https://stackoverflow.com/a/44272023/2514803)

File: generic_config_updater/gu_common.py [hunk 5]
```diff
@@ -113,9 +113,14 @@ def validate_sonic_yang_config(self, sonic_yang_as_json):
     def validate_config_db_config(self, config_db_as_json):
         sy = self.create_sonic_yang_with_loaded_models()
 
+        dup_lanes_platforms = [
```
Comment (qiluo-msft) [inline generic_config_updater/gu_common.py:116]: >dup_lanes_platforms [](http://example.com/codeflow?start=8&length=19)

you may move this block to inside `validate_lanes` function. Add a comment
```
# TODO: Move this attribute (platform with duplicated lanes in ports) to YANG models
```

Comment (qiluo-msft) [reply generic_config_updater/gu_common.py]: Could you add the TODO comment in my above comment?

**Comment by qiluo-msft** [comment]
> 

Let's remove blank line only changes, so backport will be easier.

---
Refers to: tests/generic_config_updater/files/patch_sorter_test_success.json:6161 in 6cb3a8c. [](commit_id = 6cb3a8caf9d365513e8b42085378692aecf17731, deletion_comment = False)

File: generic_config_updater/gu_common.py (hunk 5 above)
Comment (wen587) [reply generic_config_updater/gu_common.py]: Added.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**